addon/-private/system/store.js

```diff
@@ -281,6 +282,7 @@ Store = Service.extend({
     @return {DS.Model} record
   */
   createRecord(modelName, inputProperties) {
+    assert("You need to pass a type to the store's createRecord method", isPresent(modelName));
```

Comment:
Ember Data uses `modelName` for the type argument in most of these functions. I also know that `type` has also been used to refer to the same information. I think we should pick one term and be consistent. Right now I have a preference for `modelName` but I could easily be convinced to use `type` instead.